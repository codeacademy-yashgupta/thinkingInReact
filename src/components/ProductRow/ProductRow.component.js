import React, { Component } from 'react';

class ProductRow extends Component {
  render() {
    const product = this.props.product || {

    };
    let name = '';
    if(this.props.product.stocked === true){
      name = name = <span style={{color: 'black'}}>
      {product.name}
    </span>;
    }
    else if(product.stocked === false){
      name = <span style={{color: 'red'}}>
      {product.name}
    </span>;
    }
    return (
      <tr>
        <td>{name}</td>
        <td>{product.price}</td>
      </tr>
    );
  }
}

export default ProductRow;