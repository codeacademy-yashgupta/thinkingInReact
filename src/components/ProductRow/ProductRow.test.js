import React from 'react';
import ProductRow from './ProductRow.component';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('ProductRow', () => {
  const unstockedProduct={
    category: 'Sporting Goods',
    name: 'Football',
    stocked: false,
    price: 3.99
  };

  it('should render the product row', () => {
    const tree = renderer.create(<ProductRow product={unstockedProduct}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should give font in red when unstocked', () => {
    const wrapper = shallow(<ProductRow product={unstockedProduct}/>);
    expect(wrapper.find('span').props().style.color).toEqual('red');
  });
  const stockedProduct={
    category: 'Electronics',
    name: 'iPod Touch',
    stocked: true,
    price: 3.99
  };
  it('should give font in black when stocked', () => {
    const wrapper = shallow(<ProductRow product={stockedProduct} />);
    expect(wrapper.find('span').props().style.color).toEqual('black');
  })
});

