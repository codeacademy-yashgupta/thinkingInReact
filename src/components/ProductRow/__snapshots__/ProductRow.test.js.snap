// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow should render the product row 1`] = `
<tr>
  <td>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      Football
    </span>
  </td>
  <td>
    3.99
  </td>
</tr>
`;
